src/catala/catala_surface/desugaring.ml

```diff
@@ -893,6 +920,104 @@ let process_scope_use_item (precond : Ast.expression Pos.marked option)
 
 (** {1 Translating top-level items} *)
 
+(* Collects the scope definition in the set if it is an unlabeled exception *)
+let gather_unnamed_exceptions (scope : Scopelang.Ast.ScopeName.t)
+    (ctxt : Name_resolution.context)
+    (defset : Desugared.Ast.ScopeDefSet.t) 
+    (item : Ast.scope_use_item Pos.marked)
+    : Desugared.Ast.ScopeDefSet.t =
+  match Pos.unmark item with
+  | Ast.Rule rule -> (
+    match rule.Ast.rule_exception_to with
+    | Some None ->
+        let consequence_expr = Ast.Literal (Ast.LBool (Pos.unmark rule.rule_consequence)) in
+        let def =
+        {
+          Ast.definition_label = rule.rule_label;
+          Ast.definition_exception_to = rule.rule_exception_to;
+          Ast.definition_name = rule.rule_name;
+          Ast.definition_parameter = rule.rule_parameter;
+          Ast.definition_condition = rule.rule_condition;
+          Ast.definition_expr = (consequence_expr, Pos.get_position rule.rule_consequence);
+        }
+        in
+        let scope_def = get_def_key def scope ctxt in
+        Desugared.Ast.ScopeDefSet.add scope_def defset
+    | _ -> defset
+  )
+  | Ast.Definition def -> (
+    match def.Ast.definition_exception_to with
+    | Some None -> 
+      let scope_def = get_def_key def scope ctxt in
+      Desugared.Ast.ScopeDefSet.add scope_def defset
+    | _ -> defset
+  )
+  | _ -> defset
+
+let generate_default_rulenames (scope : Scopelang.Ast.ScopeName.t)
+  (ctxt : Name_resolution.context) 
+  (defset : Desugared.Ast.ScopeDefSet.t)
+  (rulemap : Desugared.Ast.RuleName.t Desugared.Ast.ScopeDefMap.t) 
+  (item : Ast.scope_use_item Pos.marked)
+  : Desugared.Ast.RuleName.t Desugared.Ast.ScopeDefMap.t =
+  let is_exception (def : Ast.definition) = match def.Ast.definition_exception_to with
+    | None -> false | _ -> true
+  in
+  match Pos.unmark item with
+  | Ast.Rule rule ->
+    let consequence_expr = Ast.Literal (Ast.LBool (Pos.unmark rule.rule_consequence)) in
+    let def =
```

Comment:
This might be refactored into `rule_to_def` since it's used in quite a number of places?

Reply:
Good point, I just pushed the refactoring